wodles/aws/tests/test_waf.py

```diff
@@ -80,10 +82,109 @@ def test_aws_waf_bucket_load_information_from_file_handles_exception_on_invalid_
+@patch('wazuh_integration.WazuhAWSDatabase.__init__')
+def test_aws_waf_bucket_check_waf_type(mock_wazuh_aws_integration, mock_sts,
+                                                   object_list: dict, result: bool):
+    """Test 'check_waf_type' method defines if the bucket contains WAF Native logs or not.
+
+    Parameters
+    ----------
+    object_list: dict
+        Objects to be returned by list_objects_v2.
+    result: bool
+        Expected result.
+    """
+    with patch('waf.AWSWAFBucket.check_waf_type'):
+        instance = utils.get_mocked_bucket(class_=waf.AWSWAFBucket)
+
+    instance.client = MagicMock()
+    instance.client.list_objects_v2.return_value = object_list
+
+    assert result == instance.check_waf_type()
+
+
+@patch('wazuh_integration.WazuhIntegration.get_sts_client')
+@patch('wazuh_integration.WazuhAWSDatabase.__init__')
+def test_aws_waf_bucket_check_waf_type_handles_exceptions(mock_wazuh_aws_integration, mock_sts):
+    """Test 'check_waf_type' handles exceptions raised and exits with the expected exit code."""
+    with patch('waf.AWSWAFBucket.check_waf_type'):
+        instance = utils.get_mocked_bucket(class_=waf.AWSWAFBucket)
+
+    with pytest.raises(SystemExit) as e:
         instance.client = MagicMock()
-        instance.client.get_object.return_value.__getitem__.return_value = f
-        instance.load_information_from_file(log_file)
+        instance.client.list_objects_v2.side_effect = Exception
+        instance.check_waf_type()
+    assert e.value.code == utils.UNEXPECTED_ERROR_WORKING_WITH_S3
+
+
+@patch('aws_bucket.AWSLogsBucket.get_base_prefix', return_value='base_prefix/')
+@patch('waf.AWSWAFBucket.check_waf_type')
+@patch('aws_bucket.AWSCustomBucket.__init__')
+def test_aws_waf_bucket_get_service_prefix(mock_custom_bucket, mock_type, mock_base_prefix):
+    """Test 'get_service_prefix' method returns the expected prefix with the format
+    <base_prefix>/<account_id>/<service>."""
+    instance = utils.get_mocked_bucket(class_=waf.AWSWAFBucket)
+
+    expected_base_prefix = os.path.join('base_prefix', utils.TEST_ACCOUNT_ID, instance.service, '')
+    assert instance.get_service_prefix(utils.TEST_ACCOUNT_ID) == expected_base_prefix
+
+
+@pytest.mark.parametrize('waf_native', [True, False])
+@patch('wazuh_integration.WazuhIntegration.get_sts_client')
+@patch('wazuh_integration.WazuhAWSDatabase.__init__')
+def test_aws_waf_bucket_get_base_prefix(mock_wazuh_aws_integration, mock_sts, waf_native: bool):
+    """Test 'get_full_prefix' method the expected base prefix depending on the WAF bucket type.
+
+    Parameters
+    ----------
+    waf_native: bool
+        Result for the 'check_waf_type' call that determines the GuardDuty bucket type.
+    """
+    with patch('waf.AWSWAFBucket.check_waf_type', return_value=waf_native):
+        instance = utils.get_mocked_bucket(class_=waf.AWSWAFBucket, prefix='prefix/')
+
+        if instance.type == "WAFNative":
```

Comment:
We could use the constants to be created here as well, or perhaps divide the tests between the native and the new ones.